cpp/src/arrow/compute/exec/util.h

```diff
@@ -57,6 +56,15 @@
 #endif
 
 namespace arrow {
+
+namespace internal {
+namespace tracing {
+
+struct Scope;
+
+}  // namespace tracing
+}  // namespace internal
+
```

Comment:
Can we instead add `Scope` to `arrow/util/type_fwd.h` (and include that here)

Reply:
Handling this in the next commit.